### PR TITLE
authentication with user memorization and redirection

### PR DESCRIPTION
**Pull request by vinyll**
Implemented:

1. App loads and detects if the user had been logged before
2. if not it show the AuthScreen (old login form) and attempts to login, then redirects to the AppScreen
3. if the loader had detected had been logged, it shows the AppScreen
4. Async storage get userID in app.js to dispatch it

To do:

- display name, avatar,…
- use `Context` to have a global state
- move the logout button to the app screen
- remove the back button arrow from the app (builtin from react-navigation, see why it's there, possibly because of the `navigate()` function call)
- fix the app to share location and toggle the button
- remove header component to [header navigation](https://reactnavigation.org/docs/en/headers.html#docsNav)

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -46,9 +46,9 @@ export default class App extends Component {
 
   render () {
     return (
-      <View style={styles.app}>
-        <HeaderApp userId={this.state.userId}/>
-        <RootStack />
+      <View style={styles.app} >
+        <HeaderApp userId={this.state.userId} />
+        <RootStack screenProps={this.state.userId} />
```
Comment (vinyll) [inline App.js]: Ohhh, is that how? Couldn't find it! 🏅

File: components/Header/AvatarApp.js [hunk 2]
```diff
@@ -7,10 +7,11 @@ import { content } from '../../static/content'
 export default class AvatarApp extends Component {
 
   render () {
-    // const initial = userName
-    //   .split(' ')
-    //   .map(string => string[0])
-    //   .join('')
+    const simulateName = 'john doe'
```
Comment (vinyll) [inline components/Header/AvatarApp.js]: Why do you need this? Can't you use a placeholder?

**Comment by vinyll** [comment]
shouldn't we stop and merge at some point?
This is an improve compare to the master anyways and we are going to use this code.

**Comment by fuentesloic** [comment]
I totally approuve 👍

**Review by fuentesloic — APPROVED** [review]

**Pull request merged**